examples/gadgets/withfilter/trace/network/network.go

```diff
@@ -0,0 +1,165 @@
+
+	err := host.Init(host.Config{
+		AutoMountFilesystems: false,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to initialized host filesystem: %w", err)
+	}
+
+	containerSelector := containercollection.ContainerSelector{}
+	containerCollection := &containercollection.ContainerCollection{}
+
+	tracerCollection, err := tracercollection.NewTracerCollection(containerCollection)
+	if err != nil {
+		return fmt.Errorf("failed to construct tracer collection: %w", err)
+	}
+
+	err = tracerCollection.AddTracer(networkTracerName, containerSelector)
+	if err != nil {
+		return fmt.Errorf("failed to add network tracer to tracer collection: %w", err)
+	}
+
+	opts := []containercollection.ContainerCollectionOption{
+		containercollection.WithTracerCollection(tracerCollection),
+		containercollection.WithContainerFanotifyEbpf(),
+		containercollection.WithLinuxNamespaceEnrichment(),
+		containercollection.WithMultipleContainerRuntimesEnrichment(
+			[]*containerutils.RuntimeConfig{
+				{Name: types.RuntimeNameDocker},
+				{Name: types.RuntimeNameContainerd},
+			}),
+		containercollection.WithPubSub(func(event containercollection.PubSubEvent) {
+			slog.Info("Container event",
+				slog.Group("runtime",
+					"containerID", event.Container.Runtime.ContainerID,
+					"containerName", event.Container.Runtime.ContainerName,
+					"containerImageName", event.Container.Runtime.ContainerImageName,
+					"containerImageDigest", event.Container.Runtime.ContainerImageDigest),
+				slog.Group("k8s",
+					"namespace", event.Container.K8s.Namespace,
+					"podName", event.Container.K8s.PodName,
+					"containerName", event.Container.K8s.ContainerName))
+		}),
+	}
+
+	networkTracer, err := tracernetwork.NewTracer()
+	if err != nil {
+		return err
+	}
+	defer networkTracer.Close()
+
+	socketEnricher, err := socketenricher.NewSocketEnricher()
+	if err != nil {
+		return err
+	}
+	defer socketEnricher.Close()
+
+	networkTracer.SetSocketEnricherMap(socketEnricher.SocketsMap())
+
+	networkTracer.SetEventHandler(func(event *tracernetworktypes.Event) {
+		if event.Type != types.NORMAL {
+			return
+		}
+
+		containerCollection.EnrichByMntNs(&event.CommonData, event.MountNsID)
+		slog.Info("Network event",
+			"timestamp", event.Timestamp,
+			"type", event.Type,
+			"pid", event.Pid,
+			"tid", event.Tid,
+			"comm", event.Comm,
+			"uid", event.Uid,
+			"gid", event.Gid,
+			"pktType", event.PktType,
+			"proto", event.Proto,
+			"port", event.Port,
+			slog.Group("destination",
+				"address", event.DstEndpoint.Addr),
+			slog.Group("runtime",
+				"runtimeName", event.Runtime.RuntimeName,
+				"containerID", event.Runtime.ContainerID,
+				"containerName", event.Runtime.ContainerName,
+				"containerImageName", event.Runtime.ContainerImageName,
+				"containerImageDigest", event.Runtime.ContainerImageDigest),
+			slog.Group("k8s",
+				"namespace", event.K8s.Namespace,
+				"podName", event.K8s.PodName,
+				"containerName", event.K8s.ContainerName))
+	})
```

Comment:
I'm wondering what's the idea of using this way to print json? To show the users all the fields that are available?